infrastructure/src/test/java/org/corfudb/infrastructure/redundancy/RedundancyCalculatorTest.java

```diff
@@ -556,4 +558,47 @@ public void testPrepareTransferWorkloadCommittedTailSplitsSegment() {
                 s -> s.equals(testSegments.get(1).getStatus()))).isTrue();
     }
 
+    @Test
+    public void testGetTrimmedNotRestoredSegments() {
+        Layout.ReplicationMode replicationMode = CHAIN_REPLICATION;
+        LayoutStripe layoutStripe1 = new LayoutStripe(ImmutableList.of("A"));
+        LayoutSegment segment1 = new LayoutSegment(replicationMode, 0L, 100L,
```

Comment:
Even if addresses before the trim mark have been trimmed, they are on disk right?  How is that data restored on B and C?

Reply:
There is no real reason to transfer trimmed addresses because logically they are trimmed. The starting address has moved and is past the trim mark. So whether they are present on disk or not, we check it either way amd get the same result:
```
private boolean isTrimmed(long address) {
        return address < dataStore.getStartingAddress();
    }
```